### PR TITLE
HPCC-28221 Metrics graph can cause ECL Watch to crash

### PR DESCRIPTION
**Pull request by GordonSmith**
Fix DockPanel render race condition with Metrics
Add ErrorBoundary utility
Improve graph ID encode/decode
Bump graph version to latest
Remove old logging
Add show fullscreen mode to metrics page

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jeclrsg rebased please review.

File: esp/src/src-react/components/Metrics.tsx [hunk 1]
```diff
@@ -15,6 +15,7 @@ import { DockPanel, DockPanelItems, ReactWidget, ResetableDockPanel } from "../l
 import { IScope, MetricGraph, MetricGraphWidget } from "../util/metricGraph";
 import { pushUrl } from "../util/history";
 import { debounce } from "../util/throttle";
+import { ErrorBoundary } from "../util/errorBoundary";
```
Comment (jeclrsg) [inline esp/src/src-react/components/Metrics.tsx:18]: Looks like this file got missed in a merge, maybe

File: esp/src/src-react/components/Metrics.tsx [hunk 2]
```diff
@@ -375,53 +404,55 @@ export const Metrics: React.FunctionComponent<MetricsProps> = ({
...
+            <ErrorBoundary>
                 <DockPanel items={items} layout={options?.layout} layoutChanged={layoutChanged} onDockPanelCreate={setDockpanel} />
                 <MetricsOptions show={showMetricOptions} setShow={setShowMetricOptions} />
-            </>}
+            </ErrorBoundary>
+        }
     />;
 };
+
+//
```
Comment (jeclrsg) [inline esp/src/src-react/components/Metrics.tsx:458]: Stray comment here?

**Review by jeclrsg — CHANGES_REQUESTED** [review]
@GordonSmith ErrorBoundary didn't get added in the rebase possibly

**Review by jeclrsg — APPROVED** [review]

**Comment by ghalliday** [comment]
@GordonSmith did this have the correct HPCC-number?  There was already a PR against that number.

**Pull request merged**